### PR TITLE
Remove C casts in RankInfo class

### PR DESCRIPTION
**Pull request by xezon**

### CODE REVIEW

File: src/game/common/ini/ini.cpp [hunk 1]
```diff
@@ -112,7 +112,7 @@ BlockParse TheTypeTable[] = {
     {"Road", &TerrainRoadCollection::Parse_Terrain_Road_Definition},
     {"Science", (iniblockparse_t)(0x00488EA0) /*&INI::parseScienceDefinition*/},
     //{ "Rank", (iniblockparse_t)(0x00489800)/*&INI::parseRankDefinition*/ },
-    {"Rank", &RankInfoStore::Parse_Rank_Definition},
+    {"Rank", &RankInfoStore::Friend_Parse_Rank_Definition},
```
Comment (duncanspumpkin) [inline src/game/common/ini/ini.cpp]: Why the name change? The old name was consistent with the other functions

Comment (xezon) [reply src/game/common/ini/ini.cpp]: This is the name I saw in original. Can you double check?

Comment (duncanspumpkin) [reply src/game/common/ini/ini.cpp]: I don't really care about original i care about consistency. The original had this function as part of ini but we just make it public so it's not a friend anymore.

Comment (xezon) [reply src/game/common/ini/ini.cpp]: There is also this:
`friend void RankInfoStore::Friend_Parse_Rank_Definition(INI *)`

I suspect that's the reason for the Friend name?

Comment (duncanspumpkin) [reply src/game/common/ini/ini.cpp]: Yes but it's unused

Comment (xezon) [reply src/game/common/ini/ini.cpp]: Removing that `friend` will break compilation. So it appears to be used.

Comment (xezon) [reply src/game/common/ini/ini.cpp]: Restored new `Parse_Rank_Definition` name.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**